jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/FilesMojoV2.java

```diff
@@ -251,7 +252,17 @@ public void execute() throws MojoExecutionException, MojoFailureException {
           Xpp3Dom paths = extraDirectoriesConfiguration.getChild("paths");
           if (paths != null) {
             // <extraDirectories><paths><path>...</path><path>...</path></paths></extraDirectories>
-            return xpp3ToList(paths, Paths::get);
+            // paths can contain either strings or ExtraDirectory objects
+            List<Path> pathList = new ArrayList<>();
+            for (Xpp3Dom path : paths.getChildren()) {
+              Xpp3Dom from = path.getChild("from");
+              if (from != null) {
+                pathList.add(Paths.get(from.getValue()));
+              } else {
+                pathList.add(Paths.get(path.getValue()));
+              }
+            }
+            return Collections.unmodifiableList(pathList);
```

Comment:
I think we can remove `unmodifiableList` for consistency. It's weird to sometimes return a modifiable list (line 242) and other times an unmodifiable one.

Reply:
I errorprone (?) complained about this same inconsistency before I added this. I thought `Collections.singletonList` and `Collectors.toList` were unmodifiable?

Reply:
Ah, `singletonList()` is immutable according to the Javadoc. And for `Collectors.toList()`, its Javadoc says,

> There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned

I think we're good.